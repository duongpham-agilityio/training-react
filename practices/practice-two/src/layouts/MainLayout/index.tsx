import { Suspense } from 'react';
import { SWRConfig } from 'swr';
import { Outlet } from 'react-router-dom';

// Layout
import Header from 'layouts/Header';

// Context
import { PopupProvider, ToastProvider } from 'contexts';

// Helpers
import { fetcher } from 'helpers';

const MainLayout = () => {
  return (
    <SWRConfig
      value={{
        fetcher,
        revalidateOnFocus: false,
      }}
    >
      <ToastProvider>
        <PopupProvider>
          <Header />
          <Suspense>
            <Outlet />
          </Suspense>
        </PopupProvider>
      </ToastProvider>
    </SWRConfig>
  );
};

export default MainLayout;
