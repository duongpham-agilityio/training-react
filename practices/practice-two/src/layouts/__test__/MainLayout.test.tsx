// Mocks
import 'components/__test__/mocks/image.test';

import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

// Components
import MainLayout from 'layouts/MainLayout';

const setup = () =>
  render(
    <BrowserRouter>
      <MainLayout />
    </BrowserRouter>
  );

describe('MainLayout', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
