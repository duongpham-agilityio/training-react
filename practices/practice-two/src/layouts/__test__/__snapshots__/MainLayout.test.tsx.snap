// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout Match to snapshot 1`] = `
<div>
  <header>
    <a
      href="/"
    >
      <img
        alt="logo"
        height="60px"
        src="assets/icons/Logo.jpg"
        width="100px"
      />
    </a>
    <h2
      class="undefined undefined "
    >
      Hello! Welcome to the Book Marker
    </h2>
  </header>
</div>
`;
